### PR TITLE
submenu for controlling multiple vehicle electronics

### LINKED ISSUE #21750: Add "setup vehicle" menu to vehicle controls along with "control vehicle"
I'd like to see additional menu choice in vehicle controls along with "control vehicle".
"setup vehicle" or any similar name, which is completely the same as "control vehicle" but does not close itself after you select an option. 

Useful in situation like: Stop driving at night, turn off engine, turn off headlights, turn on aisle lights to do some stuff before going to sleep. Right now we have to access controls multiple times to achieve all this.

### LINKED ISSUE #24166: QoL Change Vehicle controls menu to allow multiple changes before exiting.
Game version: 7520

Operating system: Windows 7 (others not tested)

Tiles or curses: (Tiles)

Mods active:

### Expected behavior
To be able to add multiple "turn off/on" requests to the vehicle control interaction menu (^ or E for examin on controls tile). For example choose to turn off a refrigerator and a recharger at the same time.

### Actual behavior
When using vehicle interaction (^ or E for examin on controls tile) to turn on or off a device, the game applies this instantly and exits the menu. If a player wishes to turn on headlights and ilse lights, or turn off fridge and recharger, they must go back into the menu for each item.

### Steps to reproduce the behavior
Add multiple electronic devices to a vehicle and activate more than one.

Currently it is working as intended, and it would only save a few button clicks, so I guess it would be more a GUI toggle thing. As cars with just headlamps/dome lights won't have long lists of items to turn on and off. But vehicles with lights, recharger, turrets, fridge, stereo etc, can get tedious to turn everything on and off when entering/exiting. So a toggle to auto exit the menu (current) or wait in menu (proposed) between selections would be helpful.

### PR DESCRIPTION
**Pull request by mlangsdorf**
add an option to the vehicle control menu that spawns a new submenu that allows all the electronic subsystems (lights, farm equipment, music, etc) to be toggled without exiting the menu until the player hits quit or escape.

this doesn't remove any of the existing options to control electronic subsystems, so the player has the option of a single keypress if they just want to toggle one thing, or they go into the subsystem if they're doing several things.

a future step will be to add some convenience toggles (fridge + stereo + recharger, headlights + floodlights, all farm equipment) but this should be a start.

Fixes #24166 
Fixes #21750

### CODE REVIEW

**Comment by AMurkin** [comment]
A menu(s) screenshots would be helpful.

File: src/vehicle.cpp [hunk 1]
```diff
@@ -607,6 +607,108 @@ void vehicle::control_doors() {
...
+        add_toggle( _( "stereo" ), keybind( "TOGGLE_STEREO" ), "STEREO" );
+        add_toggle( _( "chimes" ), keybind( "TOGGLE_CHIMES" ), "CHIMES" );
+        add_toggle( _( "fridge" ), keybind( "TOGGLE_FRIDGE" ), "FRIDGE" );
+        add_toggle( _( "recharger" ), keybind( "TOGGLE_RECHARGER" ), "RECHARGE" );
+        add_toggle( _( "plow" ), keybind( "TOGGLE_PLOW" ), "PLOW" );
+        add_toggle( _( "reaper" ), keybind( "TOGGLE_REAPER" ), "REAPER" );
+        add_toggle( _( "planter" ), keybind( "TOGGLE_PLANTER" ), "PLANTER" );
+        add_toggle( _( "scoop" ), keybind( "TOGGLE_SCOOP" ), "SCOOP" );
+        add_toggle( _( "water purifier" ), keybind( "TOGGLE_WATER_PURIFIER" ), "WATER_PURIFIER" );
+
```
Comment (Night-Pryanik) [inline src/vehicle.cpp]: No washing machine? Also, soon there will be freezer vehicle part. Have you considered moving all this to json somehow?

**Comment by Night-Pryanik** [comment]
IMO this should be the default. It's not a big deal to hit Esc after toggling one particular electronics.

File: src/vehicle.cpp (hunk 1 above)
Comment (mlangsdorf) [reply src/vehicle.cpp]: I didn't remove anything from the original code; there's no washing machine toggle before this PR and I didn't add one.

A general refactor of vehicle controls into its own class with more json support is probably a good idea but way beyond the scope of this PR.

**Comment by mlangsdorf** [comment]
Two reasons for not making this the default:
1. There are some people who actually feel it is a big deal to hit escape after toggling a particular piece of electronics, and this solution addresses that issue.
2. The logic for handling all vehicle controls in a continuous loop is fairly gnarly, because you have things like honk horn, where the player expects an immediate feedback, mixed in with electronics toggles, where he doesn't. Turrets are a separate but similar problem. Hoisting all the electronics to a separate submenu avoids that problem for now.

**Comment by mlangsdorf** [comment]
multi-controls1
The normal vehicle control menu with the new option in a Luxury RV
multi-controls2
The electronics control submenu

**Pull request merged**